src/manager.cpp

```diff
@@ -178,6 +294,127 @@ std::string
     return std::string{};
 }
 
+std::tuple<std::string, uint16_t> Manager::getUnexpandedLocationCode(
+    const std::string& i_expandedLocationCode)
+{
+    // Location code should always start with U and fulfil minimum length
+    // criteria.
+    if (i_expandedLocationCode[0] != 'U' ||
+        i_expandedLocationCode.length() <
+            constants::EXP_LOCATION_CODE_MIN_LENGTH)
+    {
+        phosphor::logging::elog<types::InvalidArgument>(
+            types::Argument::ARGUMENT_NAME("LOCATIONCODE"),
+            types::Argument::ARGUMENT_VALUE(i_expandedLocationCode.c_str()));
+    }
+
+    std::string l_kwd;
+
+    auto l_retVal = utils::readDbusProperty(
+        "xyz.openbmc_project.Inventory.Manager",
+        "/xyz/openbmc_project/inventory/system/chassis/motherboard",
+        "com.ibm.ipzvpd.VCEN", "FC");
+
+    if (auto kwdVal = std::get_if<types::BinaryVector>(&l_retVal))
+    {
+        l_kwd.assign(reinterpret_cast<const char*>(kwdVal->data()),
+                     kwdVal->size());
+    }
+
+    // get the first part of expanded location code to check for FC or TM
+    std::string l_firstKwd = i_expandedLocationCode.substr(1, 4);
+
+    std::string o_unexpandedLocationCode{};
+    uint16_t o_nodeNummber = constants::INVALID_NODE_NUMBER;
+
+    // check if this value matches the value of FC kwd
```

Comment:
Nits: Check if the value matches with FC keyword value.

Reply:
Done